utils/db/teams.js

```diff
@@ -0,0 +1,94 @@
+import shortid from "shortid";
+
+import firebase from "../firebase/firebase";
+import { setCurrentChallenge } from "./challenges";
+import { getCurrentValue } from "./helper";
+
+async function getTeamIdFromTeamCode(teamCode) {
+  return getCurrentValue(`/teamCodes/${teamCode}`);
+}
+
+export async function createTeam(teamName, userId) {
+  const team = {
+    name: teamName,
+    score: 0,
+    users: [userId]
+  };
+
+  // Create team
+  let teamRef = await firebase
+    .database()
+    .ref("/teams/")
+    .push(team);
+
+  let teamId = teamRef.key;
+
+  // Use the last six digits of team id for the team code
+  let teamCode = teamId.slice(-6);
+
+  // Race condition in the event that two users generate the same team code and check for existence at the same time
+  // Very unlikely but might want to come up with a solution for this eventually
+  let teamCodeExists = await getTeamIdFromTeamCode(teamCode);
+  while (teamCodeExists) {
+    // Delete team
+    firebase
+      .database()
+      .ref(`/teams/${teamId}`)
+      .remove();
+    // Create another team
+    teamRef = await firebase
+      .database()
+      .ref("/teams/")
+      .push(team);
+
+    teamId = teamRef.key;
+
+    // Use the last six digits of team id for the team code
+    teamCode = teamId.slice(-6);
+    teamCodeExists = await getTeamIdFromTeamCode(teamCode);
+  }
```

Comment:
Lines 39 through 47 are the same as the block used outside this race condition check to initially create the team (lines 19 through 27).

Would it be a good idea to isolate this code to a separate function (maybe called `createTeam()`) to avoid repeated code?

Reply:
Updated with changes